lib/ThreeScaleClient.php

```diff
@@ -95,7 +93,18 @@ public function getHost() {
    */
   public function authorize($appId, $appKey = null, $serviceId = null, $usage = null) {
     $url = "http://" . $this->getHost() . "/transactions/authorize.xml";
-    $params = array('provider_key' => $this->getProviderKey(), 'app_id' => $appId);
+    $params = array('app_id' => $appId);
+
+    $providerKey = $this->getProviderKey();
+    $serviceToken = $this->getServiceToken();
+
+    if($providerKey) {
+      $params['provider_key'] = $providerKey;
+    }
+
+    if ($serviceToken) {
```

Comment:
I think there is a way of "overloading" this method in PHP.

What if you could create a tuple of `<service_id, service_token>` and pass it instead of `serviceId` parameter ?

Then `serviceId` parameter could accept two types: serviceId as string or some custom type of serviceId and serviceToken. That way you'd not have to create different method.

Sorry it came so late. I guess PHP can check type of an object no? So just checking if serviceId is a string or some custom type and extract both serviceId and serviceToken from it.